utility/utility.py

```diff
@@ -11,9 +12,41 @@ def getRole(roles, roleName):
     return None
 
 
-# restricts roles that can be picked based on roles.py
-def isGamesRole(role):
-    for validRole in ROLES_LIST:
-        if validRole.name.casefold() == str(role).casefold():
+# find bot's member object in guild
+# returns None
+def getBotMember(guild, bot):
+    for m in guild.members:
+        if m == bot.user:
+            return m
+    return None
+
+
+# returns if role is found in guild, has no permissions
+# and is manageable by the bot
+def isGamesRole(guild, bot, role):
+    botMember = getBotMember(guild, bot)
+
+    if botMember is None:
+        return False
```

Comment:
Why do we need to find the bot user here? Isn't bot passed in so we have that to use?

Reply:
The utility class is purely functional. It does not have a class with a bot attribute like the cog classes do.

isGamesRole is used by both partycog and rolescog, so I had a tough time deciding to move isGamesRole out of the utility class.

Because of the nature of the utility class, it also does not know the context of its usage. That's why I had to pass in the guild and bot.

I would like to change the architecture, but I had a hard time figuring it out. Tbh I didn't give it much thought, and decided to fix it later when I'm up for it. If im going to make some fixes and you have an idea on architecture, I'm happy to make it work

Reply:
Ohhh Member is different than User in that it gives guild context

Reply:
Could we use the Discord utility for simplicity
` botMember = discord.utils.find(lambda m: m == bot.user, guild.members)`

Essentially the same as what you're defining.